lisa/tools/python.py

```diff
@@ -49,17 +49,30 @@ def _install(self) -> bool:
         self.node.os.install_packages(package_name)
         return self._check_exists()
 
-    def install_packages(self, packages_name: str) -> None:
-        cmd_result = self.run(
-            f"install -q {packages_name}",
-        )
-        if 0 != cmd_result.exit_code and get_matched_str(
-            cmd_result.stdout, self._no_permission_pattern
-        ):
+    def install_packages(self, packages_name: str, cache_dir="") -> None:
```

Comment:
the cache_dir and -b are two different folders. It's not good practice to put them in the same folder, but it's ok, if it doesn't break.

Reply:
> I tried with `redhat rhel 9_1 9.1.2022112113` it failed for
> 
> ```
> failed on run gpu script: import torch;print(f'gpu count: {torch.cuda.device_count()}'), output: Traceback (most recent call last):
> 
>   File "<string>", line 1, in <module>
> 
> ModuleNotFoundError: No module named 'torch'
> get unexpected exit code on cmd ['python3', '-c', "import torch;print(f'gpu count: {torch.cuda.device_count()}')"]] Expected <[0]> to contain item <1>, but did not
> ```

Redhat 9.1 using a different python(3.9) and pip version, which made them does not treat "-b" parameter. Changed to update TMP folder.

Reply:
> 

I added both build and cache parameters.

Reply:
It doesn't need to add both path, because the path is used to provide more space. You can call the path like "target_path", and create a path for cache, and another one for installed packages in the pip tool. So it's clear for external and internal of the pip tool both.